cmd/cel-eval/cmd/root.go

```diff
@@ -0,0 +1,174 @@
+package cmd
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/checker/decls"
+	celext "github.com/google/cel-go/ext"
+	"github.com/spf13/cobra"
+	triggerscel "github.com/tektoncd/triggers/pkg/interceptors/cel"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	corev1lister "k8s.io/client-go/listers/core/v1"
+)
+
+var (
+	rootCmd = &cobra.Command{
+		Use:   "cel-eval",
+		Short: "Tekton CEL interceptor evaluator",
+		Run:   rootRun,
+	}
+
+	expressionPath string
+	httpPath       string
+)
+
+func init() {
+	rootCmd.Flags().StringVarP(&expressionPath, "expression", "e", "", "Expression to evaluate")
+	rootCmd.Flags().StringVarP(&httpPath, "http_request", "r", "", "Path to HTTP request")
+	if err := rootCmd.MarkFlagRequired("expression"); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+	}
+	if err := rootCmd.MarkFlagRequired("http_request"); err != nil {
+		fmt.Fprintln(os.Stderr, err)
+	}
+}
+
+func rootRun(cmd *cobra.Command, args []string) {
+	if err := evalCEL(os.Stdout, expressionPath, httpPath); err != nil {
+		log.Fatal(err)
+	}
+}
+
+type secretLister struct{}
+
+func (sl secretLister) List(selector labels.Selector) (ret []*v1.Secret, err error) {
+	return nil, nil
+}
+
+func (sl secretLister) Secrets(namespace string) corev1lister.SecretNamespaceLister {
+	return secretLister{}
+}
+
+func (sl secretLister) Get(name string) (*v1.Secret, error) {
+	return nil, nil
+}
+
+func evalCEL(w io.Writer, expressionPath, httpPath string) error {
+	// Read HTTP request.
+	expression, err := readExpression(expressionPath)
```

Comment:
for CEL expressions it would be nice to read them from the CLI directly as well.

Reply:
``--expression '<expression>'`?

Reply:
yup!